src/mission_execution/nav2_mission_executor/src/mission_executor.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <exception>
+#include "nav2_mission_executor/mission_executor.hpp"
+
+using namespace std::chrono_literals;
+using nav2_tasks::TaskStatus;
+
+namespace nav2_mission_execution
+{
+
+MissionExecutor::MissionExecutor(const std::string & name)
+: nav2_tasks::ExecuteMissionTaskServer(name)
+{
+  RCLCPP_INFO(get_logger(), "MissionExecutor::MissionExecutor");
+  navigationTask_ = std::make_unique<nav2_tasks::NavigateToPoseTaskClient>("SimpleNavigator", this);
+}
+
+MissionExecutor::~MissionExecutor()
+{
+  RCLCPP_INFO(get_logger(), "MissionExecutor::~MissionExecutor");
+}
+
+TaskStatus
+MissionExecutor::execute(const nav2_tasks::ExecuteMissionCommand::SharedPtr command)
+{
+  RCLCPP_INFO(get_logger(), "MissionExecutor:execute");
+  RCLCPP_INFO(get_logger(), "MissionExecutor:execute: plan: %s",
+    command->mission_plan.c_str());
+
+  // TODO(mjeronimo): Validate the mission plan for syntax and semantics

```

Comment:
Can we track these "TODO" items as issues in Github?

Reply:
Yes, we'll carry forward any outstanding work, converting to github issues.